### PR TITLE
Parse version 2 of the ad catalog

### PR DESCRIPTION
**Pull request by moritzhaller**
Fixes: https://github.com/brave/brave-browser/issues/5231

@tmancey @jsecretan I thought about the scope for this ticket and interpreted "Parse and save" now as adding the new fields to the catalog state but not persisting it in the database. Changing the sqlite schema and bundle would be part of another ticket, since functionality is preserved and the new catalog is parsed. E.g. persisting and handling conversions is part of [ticket #6536](https://github.com/brave/brave-browser/issues/6536). Handling day parts and priority are not in scope for either of those tickets.

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ads/src/bat/ads/internal/catalog_os_info.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#ifndef BAT_ADS_INTERNAL_CATALOG_OS_INFO_H_
+#define BAT_ADS_INTERNAL_CATALOG_OS_INFO_H_
+
+#include <string>
+
+namespace ads {
+
+struct OsInfo {
+  OsInfo();
+  explicit OsInfo(const OsInfo& info);
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/catalog_os_info.h]: Please remove explicit, we will be removed from other structures as we refactor/change code

File: vendor/bat-native-ads/src/bat/ads/internal/catalog_state.cc [hunk 2]
```diff
@@ -28,7 +28,7 @@ CatalogState::~CatalogState() = default;
 Result CatalogState::FromJson(
     const std::string& json,
     const std::string& json_schema,
-    std::string* error_description) {
+    std::string* error_description) {
```
Comment (tmancey) [inline vendor/bat-native-ads/src/bat/ads/internal/catalog_state.cc]: Please revert this change as only adds 2 spaces

File: vendor/bat-native-ads/src/bat/ads/internal/catalog_os_info.h (hunk 1 above)
Comment (moritzhaller) [reply vendor/bat-native-ads/src/bat/ads/internal/catalog_os_info.h]: @tmancey I removed it from day_part_info.h as well
